### ISSUE TITLE
feat(inference): per-parameter-group learning rates + ensembled tempered MAP

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/pyrox#200]
## Summary

Two small inference-layer additions that make latent-variable models in `pyrox`
fit reliably:

1. **A `param_group_optimizer` helper** for assigning different learning rates
   to different parameter groups. Expressible today via `optax.multi_transform`,
   but not exposed and not demonstrated anywhere in the repo.
2. **Documentation + a test** establishing that `EnsembleMAP(prior_weight=1/β)`
   is the tempered-MAP objective, and that ensembling is the right answer to
   rotationally non-identifiable latent models.

Neither is a new algorithm. Both are ergonomics with real correctness
consequences.

### Part 1 — learning rates

`EnsembleMAP` and the NumPyro `SVI` paths take a single
`optax.GradientTransformation`. For any model that mixes a large block of free
latent parameters with a handful of kernel hyperparameters, one global rate is a
bad compromise:

- `Z` in `LatentFactorGPPrior` is `N×Q` free parameters on a well-conditioned,
  roughly quadratic objective — it wants a large step.
- Lengthscales and noise are a handful of parameters on a log scale with a much
  rougher landscape — a large step destabilizes them.

The GPLFR reference implementation uses a **10× ratio** (`1e-2` for latents,
`1e-3` for everything else) and dispatches on parameter name inside
`pyro.optim.Adam`. A single rate either crawls on `Z` or blows up the kernel
hyperparameters. This will bite anyone who fits issue [C]'s model, and it
generalizes to any future GP-LVM-family or deep-kernel work.

### Part 2 — tempering and ensembling

`EnsembleMAP`'s docstring already gives the tempered objective with
`prior_weight`. What is missing is the connection: **tempering the prior by
`w_prior` has the same argmax as tempering the likelihood by `β = 1/w_prior`**,
since

$$
\beta \mathcal{L} + \mathcal{P} = \beta\left(\mathcal{L} + \tfrac{1}{\beta}\mathcal{P}\right)
$$

and rescaling an objective does not move its argmax. Worth stating explicitly,
**with the caveat that they are not interchangeable in general**: `prior_weight`
scales the *entire* log prior including hyperpriors, while a likelihood-side `β`
(as in issue [C]'s `lfr_factor`) leaves priors at unit weight. Same argmax,
different gradient scaling, and different behaviour under minibatching. Callers
should know which one they want.

Separately: ensembling is not just a convenience for latent-factor models. The
`N(0,I)` decoder prior is rotation-invariant, so the objective has a
`Q(Q−1)/2`-dimensional flat manifold of optima and a `Z` landscape that is
multi-modal in practice. `E` independent MAP fits from different seeds is a
principled response, and it is something the GPLFR reference implementation has
no analogue for. This is plausibly a real accuracy improvement over that
baseline, not just parity.

### Part 1 — `param_group_optimizer`

Add to `packages/pyrox/src/pyrox/inference/`. Thin wrapper over
`optax.multi_transform`; do not reimplement its logic.

```python
def param_group_optimizer(
    groups: dict[str, optax.GradientTransformation],
    label_fn: Callable[[tuple, Any], str],
) -> optax.GradientTransformation:
    """Apply a different optimizer to each labelled parameter group.

    Models that mix a large block of free latent parameters with a few kernel
    hyperparameters usually want different step sizes for each: the latents sit
    on a well-conditioned objective and tolerate a large step, while
    lengthscales and noise live on a log scale and destabilize under one. A
    10x ratio is a common starting point.

    Args:
        groups: Maps a group label to the optimizer for that group. Every
            label returned by ``label_fn`` must be a key here.
        label_fn: Called as ``label_fn(path, leaf)`` for each parameter, where
            ``path`` is the ``jax.tree_util`` key path. Returns the group
            label for that parameter.

    Returns:
        A single `optax.GradientTransformation` suitable for
        `EnsembleMAP` or `numpyro.infer.SVI`.

    Examples:
        >>> import optax
        >>> def by_name(path, _):
        ...     return "latents" if "Z_T" in str(path) else "globals"
        >>> opt = param_group_optimizer(
        ...     {"latents": optax.adam(1e-2), "globals": optax.adam(1e-3)},
        ...     by_name,
        ... )
    """
    return optax.multi_transform(
        groups,
        param_labels=lambda tree: jax.tree_util.tree_map_with_path(label_fn, tree),
    )
```

Export from `pyrox.inference`. Add to `docs/api/inference.md`.

### Part 2 — documentation and a pinning test

No new code. Extend the `EnsembleMAP` class docstring with a short
**"Tempering"** section stating the `w_prior = 1/β` equivalence *and* the caveat
that it scales all priors. Cross-reference
[`lfr_factor`][pyrox_gp.lfr_factor] as the likelihood-side alternative.

Then add a test that pins the equivalence rather than leaving it as prose.

## Worked example (add to docs, depends on [C])

```python
import jax, optax
from numpyro.infer import SVI, Trace_ELBO
from numpyro.infer.autoguide import AutoDelta
from numpyro.infer.initialization import init_to_median, init_to_sample
from pyrox.inference import param_group_optimizer
from pyrox_gp import RBF, LatentFactorGPPrior, lfr_model

Q = 6
prior = LatentFactorGPPrior(
    kernels=tuple(RBF(pyrox_name=f"RBF_q{q}", input_dim=X.shape[1]) for q in range(Q)),
    X=X,
)

# Latents take a 10x larger step than the kernel hyperparameters.
optimizer = param_group_optimizer(
    {"latents": optax.adam(1e-2), "globals": optax.adam(1e-3)},
    lambda path, _: "latents" if "Z_T" in str(path) else "globals",
)

guide = AutoDelta(
    lfr_model,
    init_loc_fn=lambda site: (
        init_to_sample(site) if site["name"] == "Z_T" else init_to_median(site)
    ),
)
result = SVI(lfr_model, guide, optimizer, loss=Trace_ELBO()).run(
    jax.random.PRNGKey(0), 5000, X, Y, prior
)
```

## Acceptance criteria

- [ ] **Groups get different updates.** Two-group problem where one group's
      optimizer is `optax.set_to_zero()`; assert those parameters are unchanged
      after a step and the others are not.
- [ ] **Learning rate ratio is honoured.** Two groups with rates differing by
      10×, one step from identical gradients; assert the update magnitudes
      differ by the expected ratio.
- [ ] **Single group is a no-op wrapper.** With one group, updates match calling
      that optimizer directly, to `1e-12`.
- [ ] **Unlabelled parameter fails loudly.** A `label_fn` returning a key absent
      from `groups` raises with a message naming the missing label — not a
      silent skip.
- [ ] **Works inside `EnsembleMAP`.** Pass it as `optimizer=`; run a short fit
      on the existing `linear_problem` fixture in
      `tests/inference/test_ensemble.py`; assert finite loss and that both
      groups moved.
- [ ] **Tempering equivalence is pinned.** On a small conjugate problem with a
      closed-form or numerically-converged optimum, assert `prior_weight=1/β`
      and an explicit likelihood-side `β` (via `numpyro.handlers.scale`) reach
      the same argmax to a stated tolerance. Mark `@pytest.mark.slow`.
- [ ] **Docstring updated.** `EnsembleMAP` gains a "Tempering" section covering
      both the equivalence and the all-priors caveat.
- [ ] **Docs.** `param_group_optimizer` in `docs/api/inference.md`; the worked
      example lands wherever [C]'s narrative docs go.

## Out of scope

- Learning-rate schedules — `optax.multi_transform` composes with schedules
  already; do not wrap that
- Any new optimizer
- Automatic group inference. The `label_fn` is explicit on purpose; guessing
  which parameters are "latents" from a pytree is exactly the kind of
  speculative flexibility `AGENTS.md` warns against
- Changing any existing default learning rate

## Before reporting done

```bash
uv run pytest -q
uv run --group lint ruff check .
uv run --group lint ruff format --check .
uv run --group typecheck ty check packages/pyrox/src/pyrox
```
